examples/notebooks/beam-ml/rag_usecase/redis_connector.py

```diff
@@ -0,0 +1,344 @@
+            self.client.close()
+
+
+"""This module implements IO classes to read write text Embeddings in Redis.
+
+
+Insert Embedding in Redis :
+-----------------
+:class:`InsertEmbeddingInRedis` is a ``PTransform`` that writes key and values to a
+configured sink, and the write is conducted through a redis pipeline.
+
+The ptransform works by getting the first and second elements from the input,
+this means that inputs like `[k,v]` or `(k,v)` are valid.
+
+Example usage::
+
+  pipeline | InsertEmbeddingInRedis(host='localhost',
+                          port=6379,
+                          batch_size=100)
+"""
+
+
+class InsertEmbeddingInRedis(PTransform):
+    """WriteToRedis is a ``PTransform`` that writes a ``PCollection`` of
+    key, value tuple or 2-element array into a redis server.
+    """
+
+    def __init__(self,
+                 host: str,
+                 port: int,
+                 command: Optional[str] = None,
+                 batch_size: int = 100,
+                 embedded_columns: list = []
+                 ):
+
+        """
+
+        Args:
+        host (str): The redis host
+        port (int): The redis port
+        command (str): command to be executed with redis client
+        batch_size (int): Number of key, values pairs to write at once
+        embedded_columns (list): list of column whose embedding needs to be generated
+
+        Returns:
+        :class:`~apache_beam.transforms.ptransform.PTransform`
+
+        """
+
+        self._host = host
+        self._port = port
+        self._command = command
+        self._batch_size = batch_size
+        self.embedded_columns = embedded_columns
+
+    def expand(self, pcoll):
+        return pcoll \
+               | "Reshuffle for Embedding in Redis Insert" >> Reshuffle() \
+               | "Write `Embeddings` to Redis" >> beam.ParDo(_WriteEmbeddingInRedisFn(self._host,
+                                                                                      self._port,
+                                                                                      self._command,
+                                                                                      self._batch_size,
+                                                                                      self.embedded_columns))
+
+
+class _WriteEmbeddingInRedisFn(DoFn):
+    """Abstract class that takes in redis  credentials
+    to connect to redis DB
+    """
+
+    def __init__(self, host, port, command, batch_size, embedded_columns):
+        self.host = host
+        self.port = port
+        self.command = command
+        self.batch_size = batch_size
+        self.embedded_columns = embedded_columns
+
+        self.batch_counter = 0
+        self.batch = list()
+
+    def finish_bundle(self):
+        self._flush()
+
+    def process(self, element, *args, **kwargs):
+        self.batch.append(element)
+        self.batch_counter += 1
+        if self.batch_counter == self.batch_size:
```

Comment:
Nit: it's generally better to make these `self.batch_counter >= self.batch_size`

Reply:
Yeah that makes sense. I'll make the required change.